tools/git-version-gen

```diff
@@ -10,8 +10,9 @@ if [ $# -ne 1 ]; then
 fi
 
 generate() {
+	commit=$(git rev-parse --short=8 HEAD)
 	if desc=$(git -C "$BASE/.." describe); then
-		echo "$desc" | sed 's/-[0-9]\+-g.*/.x/'
+		echo "$desc" | sed "s/-[0-9]\+-g.*/.$commit/"
```

Comment:
It would seem easier to me to just let `describe` do its thing, and replace - with . to stay compatible with distro version formats. I. e. `"echo ${desc//-/.}"` (that is a bashism) or `echo "$desc" | tr - .`. So in fact the entire file could  be collapsed into this single command in configure.ac.

Reply:
right, that would work :)
Changed it, not it produces rpm like `cockpit-214.1.47.g205cf5836-1.wip.fc31.x86_64.rpm`